.evergreen/config.yml

```diff
@@ -0,0 +1,176 @@
+##############################################
+# Evergreen Template for Drivers Atlas Testing
+##############################################
+
+# Run previous commits to pinpoint a failure's origin.
+stepback: true
+
+# Mark failures other than test failures with a purple box.
+command_type: system
+
+# Limit maximum test running time.
+exec_timeout_secs: 7200   # 2 hours
+
+# What to do when evergreen hits the timeout (`post:` tasks are run automatically)
+timeout:
+  - command: shell.exec
+    params:
+      script: |
+        ls -la
+
+functions:
+  "install astrolabe":
+    # Clone astrolabe.
+    - command: git.get_project
+      params:
+        directory: astrolabe-src
+    # Setup global expansions.
+    - command: shell.exec
+      params:
+        script: |
+          export ASTROLABE_VIRTUALENV_NAME="astrolabevenv"
+          export ASTROLABE_BINARY="astrolabe-src/$ASTROLABE_VIRTUALENV_NAME/bin/astrolabe"
+          cat <<EOT > astrolabe-expansion.yml
+          PREPARE_SHELL_COMMON: |
+            set -o errexit
+            export SYSTEM_PYTHON3_BINARY="${SYSTEM_PYTHON3_BINARY}"
+            export ASTROLABE_VIRTUALENV_NAME="$ASTROLABE_VIRTUALENV_NAME"
+            export ASTROLABE_BINARY="$ASTROLABE_BINARY"
+            export ATLAS_PROJECT_NAME="${project}"
+            export EVERGREEN_BUILD_ID="{build_id}"
+            export ATLAS_API_USERNAME="${atlas_key}"
+            export ATLAS_API_PASSWORD="${atlas_secret}"
+          EOT
+          # Don't print this as it contains credentials.
+    - command: expansions.update
+      params:
+        file: astrolabe-expansion.yml
+    # Use script to install astrolabe.
+    - command: shell.exec
+      params:
+        working_dir: astrolabe-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          sh .evergreen/install-astrolabe.sh
+
+  "install driver":
+    # Clone driver source code.
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          git clone --recursive --branch "${DRIVER_REVISION}" "${DRIVER_REPOSITORY}" driver-src
+    # Make the driver's *.sh files executable.
+    - command: shell.exec
+      params:
+        working_dir: driver-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          for i in $(find .evergreen/atlas-testing -name \*.sh); do
+            chmod +x $i
+          done
+    # Setup driver-specific environment expansions.
+    - command: subprocess.exec
+      params:
+        working_dir: driver-src
+        binary: .evergreen/atlas-testing/write-driver-expansion.sh
```

Comment:
Is this necessary? I'm trying to think with my "how do we get this done for 12 drivers" hat here, could we fold both of these scripts into something like `install-dependencies.sh` which _always_ writes a `driver-expansions.yaml` file, even if its empty

Reply:
Good idea - the install-driver and write-driver-expansions scripts should be able to be combined without any issues. I will double-check before committing to the change though.

Reply:
Actually, on a closer look, it will not be possible to combine these two files into a single file. The reason being that the install-driver.sh step might _require_ some of the expansions created by the write-driver-expansion.yml step. Unfortunately, this means that we must call the expansions.update evergreen command against the YML file written by the script **before** invoking install-driver.sh.

Reply:
Scratch that, I managed to rework things so that the driver can be installed using a single script.